proteus/cpostprocessing.pyx

```diff
@@ -0,0 +1,1298 @@
+                                            <double*>(elementResidual.data),
+                                            <double*>(vAverage.data),
+                                            <double*>(dx.data),
+                                            <double*>(w.data),
+                                            <double*>(n.data),
+                                            &nodeStarFactor.nsf,
+                                            <double*>(conservationResidual.data),
+                                            <double*>(vConservative.data),
+                                            <double*>(vConservative_element.data))
+
+def calculateConservationResidualPWL_primative(np.ndarray interiorElementBoundaries,
+                                               np.ndarray exteriorElementBoundaries,
+                                               np.ndarray elementBoundaryElements,
+                                               np.ndarray elementBoundaryLocalElementBoundaries,
+                                               np.ndarray skipflag_elementBoundaries,
+                                               np.ndarray elementResidual,
+                                               np.ndarray dx,
+                                               np.ndarray n,
+                                               np.ndarray conservationResidual,
+                                               np.ndarray vConservative):
+    pp.calculateConservationResidualPWL_primative(dx.shape[0],
+                                                  interiorElementBoundaries.shape[0],
+                                                  exteriorElementBoundaries.shape[0],
+                                                  dx.shape[1],
+                                                  dx.shape[2],
+                                                  elementResidual.shape[1],
+                                                  n.shape[2],
+                                                  <int*>(interiorElementBoundaries.data),
+                                                  <int*>(exteriorElementBoundaries.data),
+                                                  <int*>(elementBoundaryElements.data),
+                                                  <int*>(elementBoundaryLocalElementBoundaries.data),
+                                                  <int*>(skipflag_elementBoundaries.data),
+                                                  <double*>(elementResidual.data),
+                                                  <double*>(dx.data),
+                                                  <double*>(n.data),
+                                                  <double*>(conservationResidual.data),
+                                                  <double*>(vConservative.data))
+
+def calculateConservationJacobianPWL_opt(int nNodes_owned,
+                                         np.ndarray interiorElementBoundaries,
+                
```

Comment:
@cekees - If the Cython directives `#cython: boundscheck=False, wraparound=False, nonecheck=False, initializedcheck=False` were inserted at the beginning of the file that would speed up these loops.

Reply:
I'm going to hold off on these for  now. In most  of the legacy stuff the looping is  done in C anyway.